include/libcyphal/presentation/presentation.hpp

```diff
@@ -112,7 +115,30 @@ class Presentation final : private detail::IPresentationDelegate
         : memory_{memory}
         , executor_{executor}
         , transport_{transport}
+        , unreferenced_nodes_{&unreferenced_nodes_, &unreferenced_nodes_}
+    {
+        unref_nodes_deleter_callback_ = executor_.registerCallback([this](const auto&) {
+            //
+            destroyUnreferencedNodes();
+        });
+        CETL_DEBUG_ASSERT(unref_nodes_deleter_callback_, "Should not fail b/c we pass proper lambda.");
+    }
+
+    Presentation(const Presentation&)                = delete;
+    Presentation(Presentation&&) noexcept            = delete;
+    Presentation& operator=(const Presentation&)     = delete;
+    Presentation& operator=(Presentation&&) noexcept = delete;
+
+    ~Presentation()
```

Comment:
should this be virtual? I'm not sure TBH.

Reply:
`Presentation` is `final`, so I don't think we need `virtual`.